src/lib/components/Searchbar/SearchBar.jsx

```diff
@@ -0,0 +1,162 @@
+    return {
+      APP,
+      LOC,
+      appColor: APP.appColor,
+      searchBarColor: APP.searchBarColor,
+      preparedLayers: MAP.layers,
+      ...ownProps
+    };
+  };
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      inputText: '',
+      selectedLayerId: null,
+    }
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleSearchInput = this.handleSearchInput.bind(this);
+  }
+
+  componentDidUpdate() {
+    const { selectedLayerId } = this.state;
+    const { preparedLayers, openCategoryForSharedLayers } = this.props;
+
+    // open menu when layer is loaded visibility is turned on
+    if (selectedLayerId && preparedLayers[selectedLayerId].visible) {
+      const toOpenLayer = {
+        id: selectedLayerId,
+        isCatOpen: false,
+      };
+      openCategoryForSharedLayers([toOpenLayer]);
+      this.setState({selectedLayerId: null});
+    }
+  }
+
+  // load layer to map
+  toggleLayer = (layer) => {
+    const { mapId, APP, LOC } = this.props;
+    if (!mapId) {
+      return null;
+    }
+    pushLayerToURL(layer, mapId);
+    this.props.dispatch(Actions.toggleLayer(mapId, layer.id));
+    const {center, zoom } = lngLat(LOC, APP);
+    if (layer.zoomOnToggle && layer.visible) {
+        window.maps.forEach((e) => {
+          e.easeTo({
+            center,
+            zoom,
+          })
+        });  
+      } 
+    // Prepare layer if layer had not been loaded
+    if (!layer.loaded && !layer.isLoading) {
+      prepareLayer(mapId, layer, this.props.dispatch);
+    }
+  }
+  
+  // make the matching part bold
+  boldQuery(indicator, query, id){
+    const indicatorToUpper = indicator.toUpperCase();
+    const queryToUpper = query.toUpperCase();
+    if (!indicatorToUpper.includes(queryToUpper)) {
+      return null;
+    }
+    const matchIndex = indicatorToUpper.indexOf(queryToUpper);
+    const querryLen = queryToUpper.length;
+
+    return (
+      <a onClick={e => this.OnsearchResultClick(e, id)}>
+        {indicator.substr(0,matchIndex)}<b>{indicator.substr(matchIndex, querryLen)}</b>{indicator.substr(matchIndex+querryLen)}
+      </a>
+    )
+  }
+
+  // find part of layer labels matching querry
+  handleSearchInput(e) {
+    this.setState({inputText: e.target.value})
+    const { handleSearchInput, preparedLayers } = this.props;
+    let input = e.target.value;
+    input = input.replace(/\s+/g, ' ')
+    input = input.trimStart()
+    const searchResults = [];
+    Object.keys(preparedLayers).forEach(key => {
+        const { label } = preparedLayers[key];
+        const result = this.boldQuery(label, input, key)
+        if (result) {
+          searchResults.push(
+            <li key={key} className="search-sector">{result}</li>
+          )
+        }
+    })
+    handleSearchInput(searchResults, input);
+  }
+
+  // called when search result is selected
+  OnsearchResultClick(e, id) {
```

Comment:
Should we have this method name in lower camel case instead of upper camel case?